docs/src/faq.md

```diff
@@ -0,0 +1,27 @@
+# FAQ
+
+## Why do I get "missing field X" error when running `install`?
+
+This error indicates one of your dependencies has an invalid manifest file. The
+error message should indicate which dependency is broken and why:
+
+```
+Error:
+   0: Failed to process dependency foo-bar@0.0.1
+   1: Failed to parse the manifest
+   2: TOML parse error at line 1, column 1
+   2:   |
+   2: 1 | [package]
+   2:   | ^
+   2: missing field `dependencies`
+```
+
+Since the CLI won't allow you to publish a package with an invalid manifest,
+you may wonder why some published packages may be broken in this way. The
+reason is that all fields used to be optional -- `impl` packages didn't need a
+`package` section, and `lib` packages didn't require `dependencies`. When the
+lockfile functionality was first introduced, both sections were made mandatory
+for all package types, which broke parsing of older manifests.
+
+To fix this you must depend on a newer version of the dependency that publishes
```

Comment:
have you thought about storing a version in the buffrs manifest so that you can indicate incompatibility via something other than failure to deserialise?

Reply:
@mara-schulke what do you think? I'm in favour of this, though we probably won't start using it seriously until 1.0.

Reply:
We are storing a version in the manifest?

Reply:
I think James was talking about storing a separate "manifest specification version" (similar to how cargo has `edition` and protobuf has the syntax version).

Reply:
Regardless I think this is out of scope for this PR. We can create a separate issue to track.

Reply:
Yes - it's typical to store a 'manifest version' in the manifest so compatibility can be flagged if breaking changes are made. This avoids semantic misinterpretation, making it easy to ahead-of-time surface "buffrs is out of date" rather than surfacing "some field is missing".

e.g. store

```
manifest-version: "1.2"
```

and if buffrs reads a manifest it can't interpret (because it uses some incompatible feature) you can error that the manifest is out of date.

Reply:
Created #64 to track / for further discussion.

Reply:
Agreed, we can think about this! But its to early to define a stable version for the manifest though